mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,383 @@
+
+
+def split_score_annotations(fhandle: TextIO):
+    """Load haydn op20 score and annotations divided.
+
+    Args:
+        fhandle (str or file-like): path to hrm annotations
+
+    Returns:
+        music21.stream.Score: score in music21 format
+        List[Tuple[Any, Any]]: roman numerals
+    """
+    score = music21.converter.parse(fhandle.name, format="humdrum")
+
+    rna = {rn.offset: rn for rn in list(score.flat.getElementsByClass("RomanNumeral"))}
+    score.remove(rna, recurse=True)
+    rna_clean = [(offset, rn) for offset, rn in rna.items() if rn]
+    return score, rna_clean
+
+
+@io.coerce_to_string_io
+def load_score(fhandle: TextIO):
+    """Load haydn op20 score with annotations from a file with music21 format (music21.stream.Score).
+
+    Args:
+        fhandle (str or file-like): path to score
+
+    Returns:
+        music21.stream.Score: score in music21 format
+
+    """
+    score, rna = split_score_annotations(fhandle)
+    return score
+
+
+def load_key_base(fhandle, resolution):
+    _, rna = split_score_annotations(fhandle)
+    annotations = []
+    for offset, rn in rna:
+        time = int(round(float(offset * resolution)))
+        tonicizedKey = rn.secondaryRomanNumeralKey
+        key = tonicizedKey or rn.key
+        annotations.append({"time": time, "key": key})
+    return annotations
+
+
+@io.coerce_to_string_io
+def load_key(fhandle: TextIO, resolution=28):
+    """Load haydn op20 key data from a file
+
+    Args:
+        fhandle (str or file-like): path to key annotations
+
+    Returns:
+        KeyData: loaded key data
+
+    """
+    keys = load_key_base(fhandle, resolution)
+    start_times, end_times, key_names = [0], [], [str(keys[0]["key"]).replace("-", "b")]
+    for ii, k in enumerate(keys):
+        if str(k["key"]).replace("-", "b") != key_names[-1]:
+            end_times.append(keys[ii]["time"] - 1)
+            start_times.append(keys[ii]["time"])
+            key_names.append(str(keys[ii]["key"]).replace("-", "b"))
+    end_times.append(keys[-1]["time"])
+    return KeyData(np.array([start_times, end_times]).astype(float).T, key_names)
+
+
+@io.coerce_to_string_io
+def load_key_music21(fhandle: TextIO, resolution=28):
+    """Load haydn op20 key data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to key annotations
+
+    Returns:
+        List[dict]: musical key data and relative time (offset (Music21Object.offset) * resolution).
+
+    """
+    return load_key_base(fhandle, resolution)
+
+
+@io.coerce_to_string_io
+def load_midi_path(fhandle: TextIO):
+    """Load path to midi file of haydn op20 musical piece
+
+    Args:
+        fhandle (str or file-like): path to score file
+
+    Returns:
+        str: midi file path
+
+    """
+    midi_path = os.path.splitext(fhandle.name)[0] + ".midi"
+    if not os.path.exists(midi_path):
+        score, _ = split_score_annotations(fhandle)
+        score.write("midi", fp=midi_path)
+    return midi_path
```

Comment:
This function is a bit confusing for me. Having "load" in the name tells me this is loading data from a file, but it looks like this is actually creating a midi file. Why creating a midi file needed?

Reply:
We was speaking about it and many people in MIR community doesn't work over scores, they prefer pianoroll or midi. That is why we decided to add a midi getter in the haydn_op20 class :)